TheMovieDatabase/ViewControllers/Favorites/View/FavoritesViewController.swift

```diff
@@ -12,47 +12,84 @@ final class FavoritesViewController: UIViewController {
 
     // MARK: - Properties
 
-    private let service = MoviesLoadingService()
+    private let accountId = UserDefaults.standard.accountId
+    private let moviesLoadingService = MoviesLoadingService()
+    private let profileService = ProfileService()
     private var movies: [Movie] = []
+    private var wasShown = false
 
     // MARK: - Subviews
 
     @IBOutlet weak private var tableView: UITableView!
     @IBOutlet weak private var blankImage: UIImageView!
     @IBOutlet weak private var blankTitle: UILabel!
+    @IBOutlet weak private var activityIndicator: UIActivityIndicatorView!
 
     // MARK: - UIViewController
 
     override func viewDidLoad() {
         super.viewDidLoad()
         configureView()
+        loadFavoriteMovies()
     }
 
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(true)
-        movies = service.getFavoriteMovies()
-        if movies.isEmpty {
-            tableView.isHidden = true
-            blankImage.isHidden = false
-            blankTitle.isHidden = false
+        if wasShown {
+            movies = moviesLoadingService.getFavoriteMovies()
+            if movies.isEmpty {
+                setEmptyState(hidden: false)
+            } else {
+                setEmptyState(hidden: true)
+            }
+            tableView.reloadData()
         } else {
-            tableView.isHidden = false
-            blankImage.isHidden = true
-            blankTitle.isHidden = true
+            wasShown = true
         }
```

Comment:
Why can't you call it in viewDidLoad? Why do you need to use `wasShown` if you have `viewDidLoad` that called once

Reply:
explained in tlgrm